utility/src/main/java/org/oppia/android/util/datetime/BUILD.bazel

```diff
@@ -0,0 +1,19 @@
+"""
+General purposes utilities for Android accessibility checking.
+"""
+
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_android_library")
+
+kt_android_library(
+    name = "date_time_util",
+    srcs = [
+        "DateTimeUtil.kt",
+    ],
+    manifest = "//utility/src/main/AndroidManifest.xml",
+    resource_files = ["//utility/src/main/res/values/strings.xml"],
```

Comment:
Note that we'll want to remove the manifest/resources from the top-level utility library as well to make sure we aren't duplicating the resources reference.

Can we actually move the manifest/resource files to be under datetime, or does that break Gradle?

Reply:
Regarding moving manifest/res files under datetime

I think it's not a good practice as this alters the project architechture

And I did try to move the strings.xml to util/datetime and copied AndroidManifest.xm lto util/datetime.xml when I build it gives the following error

```bash
ERROR: /home/farees/opensource/oppia-android/utility/src/main/java/org/oppia/android/util/datetime/BUILD.bazel:7:19: in resource_files attribute of android_library rule //utility/src/main/java/org/oppia/android/util/datetime:date_time_util_base: 'utility/src/main/java/org/oppia/android/util/datetime/strings.xml' is not in the expected resource directory structure of <resource directory>/{anim,animator,color,drawable,font,interpolator,layout,menu,mipmap,navigation,raw,transition,values,xml}/<file>. Since this rule was created by the macro 'kt_android_library', the error might have been caused by the macro implementation
ERROR: Analysis of target '//utility/src/main/java/org/oppia/android/util/datetime:date_time_util_kt' failed; build aborted: Analysis of target '//utility/src/main/java/org/oppia/android/util/datetime:date_time_util_base' failed
INFO: Elapsed time: 1.276s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (27 packages loaded, 731 targets configured)

```

Reply:
Try copying the strings.xml to datetime/res/values/strings.xml

Reply:
@fsharpasharp How do I create a `res` package inside `datetime`

Reply:
and how do I make changes for `AndroidManifest.xml`

Reply:
I tried to create a package `res.values` and copied strings.xml here 
and as for AndroidManifest.xml I copied it to `util/datetime`

but how to make DateTimeUtil use the string resources i.e, from `util/datetime/res/values/strings.xml`

when I run it gives the following error

```bash
ERROR: /home/farees/opensource/oppia-android/utility/src/main/java/org/oppia/android/util/datetime/BUILD.bazel:7:19: KotlinCompile //utility/src/main/java/org/oppia/android/util/datetime:date_time_util_kt { kt: 1, java: 0, srcjars: 0 } for k8 failed: (Exit 1): builder failed: error executing command bazel-out/host/bin/external/io_bazel_rules_kotlin/src/main/kotlin/builder ... (remaining 1 argument(s) skipped)
utility/src/main/java/org/oppia/android/util/datetime/DateTimeUtil.kt:4:31: error: unresolved reference: R
import org.oppia.android.util.R
                              ^
Compilation failure: compile phase failed:
/home/farees/.cache/bazel/_bazel_farees/be6e0c0a55083b3b0484cd1169352d74/execroot/__main__/utility/src/main/java/org/oppia/android/util/datetime/DateTimeUtil.kt:4:31: error: unresolved reference: R
import org.oppia.android.util.R
                              ^
INFO: Elapsed time: 2.625s, Critical Path: 2.08s
INFO: 2 processes: 2 internal.

```

Reply:
Hmm. It actually is preferred practice in Bazel projects to have resources as close to the location as possible, but the R file import bit is going to be a problem.

@fsharpasharp it might make more sense for us to wait to modularize resources until after all the build packages are done, and after we remove Gradle since it probably isn't going to like that. It might mean that we need to introduce explicit resource libraries at the top level to reference everywhere, and I'm not sure how that's going to work with databinding in app module yet. WDYT?